src/symbol_table/symbol_table.cpp

```diff
@@ -0,0 +1,86 @@
+#include "symbol_table/symbol_table.h"
+
+namespace jucc::symbol_table {
+
+bool LinkedList::IsEmpty() { return this->head_ == nullptr; }
+
+Node *LinkedList::CreateNewNode(std::string it_, std::string dt_, int nt_) {
+  Node *node = new Node(std::move(it_), std::move(dt_), nt_);
+  return node;
+}
+
+void LinkedList::AddNewNode(std::string it_, std::string dt_, int nt_) {
+  Node *node = LinkedList::CreateNewNode(std::move(it_), std::move(dt_), nt_);
+  node->next_ = head_;
+  head_ = node;
+}
```

Comment:
R-value references!! possibility of memory leaking.

Take input with && directly.

Reply:
std::move solves that purpose. 👀

Reply:
it solves nothing!! Creates a problem if not handled it_ and dt_ properly in the rest of the code. 
Just do make sure they are untouched after AddNewNode invocation.

Reply:
The function ends on line 16. How can they be used afterwards 👀

Reply:
I was talking abt the parent scope.
Btw, after rechecking I feel it should fine. No worries :+1: